File: Algo/src/com/schoenmaeker/Algo_3_2.java
package com.schoenmaeker;

//import java.util.Scanner;

/**
 * @author Schoenmaeker
 * @version 0.0.1
 * @since 2020
 *
 * //----------------------------------------------------//
 * // Rechercher une lettre dans une chaine de caractères
 * 
 * début
 * //chaine <-- "."
 * cpt_occ <-- 0 // compteur occurences
 * l  <-- longeur(chaine)
 * 
 * Pour i = 0 à l
 *    lettre <-- chaine[i]
 *    Si lettre = "a"
 *        cpt_occ++ // compteur occurences
 *    fin Si
 * fin Pour
 * 
 * si chaine = "."
 *    écrire "LA CHAINE EST VIDE"
 * Sinon
 *    écrire "Le nombre de a contenu dans la chaine est de ", cpt_occ
 * fin Sinon

 * fin
 */

public class Algo_3_2 {
	
	/**
	 * @param args
	 */
	// Exercice le périmétre d'un rectangle
	public static void main(String[] args) {
		
		String str = "abc.";
		int cpt_occ = 0; // compteur occurences
		int longeur_str = str.length();
		char letter;
		
		for(int i = 0; i < longeur_str ; i++ ) {
			letter = str.charAt(i);
			if(letter == 'a') {
				cpt_occ++;
			}
		}
		
		if(str == ".") {
			System.out.println("LA CHAINE EST VIDE");
		}else {
			System.out.println("Le nombre de a contenu dans la chaine est de "+cpt_occ);
		}
		
	}
	
}

File: FOAD_Javascript/Algo_JS/Algo_3_2.js

var str = "abc.";
var cpt_item = 0; // compteur occurences
var longeur_str = str.length;
var letter;
var search = "a";
var lastCharacter = str[longeur_str-1];

for(var i = 0; i < longeur_str ; i++ ) {
    letter = str.charAt(i); // str[i]
    if(letter === search) {
        cpt_item++;
    }
}

if(str == ".") {
    console.log("LA CHAINE EST VIDE");
}else {
    console.log("Le nombre de a contenu dans la chaine est de ",cpt_item);
}